boards.txt

```diff
@@ -439,6 +439,41 @@ Disco.menu.upload_method.STLink.upload.tool=stlink_upload
 ################################################################################
 # Other boards
 
+###############################
+# Generic F030
+GenF030.name=Generic STM32F030 series
+
+GenF030.build.vid=0x1EAF
+GenF030.build.pid=0x0004
+GenF030.vid.0=0x1EAF
+GenF030.pid.0=0x0004
+
+GenF030.build.core=arduino
+GenF030.build.board=GenF030
+GenF030.build.extra_flags=-D{build.product_line} {build.enable_usb} {build.xSerial}
+
+# Generic F030 board
+# Support: Serial1 (USART1 on PA10, PA9)
+GenF030.menu.pnum.GENERIC_F030K6=Generic F030K6
```

Comment:
If the idea is to have only one entry and then select the proper RAM/flash size and appropriate linker script, then it should be rewrote:

Something like that:
GenF030.menu.pnum.GENERIC_F030xx=Generic F030xx

and adding a new menu entry:
menu.var=Variant

Then: 
GenF030.menu.var.K6=K6
GenF030.menu.var.K6.upload.maximum_size=32768
GenF030.menu.var.K6.upload.upload.maximum_data_size=4096
GenF030.menu.var.K6.build.board=GENERIC_F030K6
GenF030.menu.var.K6.build.ldscript=ldscriptK6.ld
GenF030.menu.var.K6.build.product_line=STM32F030x6

So new var could be added like that:
GenF030.menu.var.XY=XY
GenF030.menu.var.XY.upload.maximum_size=xxxxx
GenF030.menu.var.XY.upload.upload.maximum_data_size=xxxx
GenF030.menu.var.XY.build.board=GENERIC_F030XY
GenF030.menu.var.XY.build.ldscript=ldscriptXY.ld
GenF030.menu.var.XY.build.product_line=STM32F030xY

I did not test but the philosophy is here, I guess

Reply:
I have tried to understand your idea. It would be great if we could choose STM32F030xx as Chip Series and then slecet at the "Board part Number" the appropriate chip. 
Unfortunately i was not able to get this working... 

i see that you want to replace the menu.var with the variant. But how can you choose the variant variable?

Reply:
Arduino menu is not very convenient.
The menu var is only a new submenu where you can select the chip version (K6, K8, R8,...)

Reply:
Ok i think i see your goal. 
I will try to implement it. 
Thanks